File: bank-app/app/src/main/java/com/bankapp/loginScreen/LoginRouter.java
package com.bankapp.loginScreen;


import android.content.Intent;

import androidx.annotation.NonNull;

import com.bankapp.helper.ConstantsHelper;
import com.bankapp.statementScreen.StatementActivity;

import java.lang.ref.WeakReference;


interface LoginRouterInput {
    public void passDataToNextScene();
    public Intent determineNextScreen();
}

public class LoginRouter implements LoginRouterInput {

    public static String TAG = LoginRouter.class.getSimpleName();

    public WeakReference<LoginActivity> activity;

    public void passDataToNextScene(Intent intent) {
        intent.putExtra(ConstantsHelper.USER_DETAILS, activity.get().userAccount);
    }

    @NonNull
    @Override
    public Intent determineNextScreen() {
        //TODO: Login - Your rules to next screen here
        return new Intent(activity.get(), StatementActivity.class);
    }

    @Override
    public void passDataToNextScene() {
        Intent intent = determineNextScreen();
        passDataToNextScene(intent);
        activity.get().startActivity(intent);
        activity.get().finish();
    }
}

File: BankApp/app/src/main/java/com/schaefer/bankapp/login_screen/router/LoginRouter.kt
package com.schaefer.bankapp.login_screen.router

import android.content.Intent
import com.schaefer.bankapp.login_screen.activity.LoginActivity
import java.lang.ref.WeakReference

class LoginRouter : CurrencyRouterInput {
    var activity: WeakReference<LoginActivity>? = null
    override fun passDataToNextScene(intent: Intent) {
        activity?.get()?.startActivity(intent)
        activity?.get()?.finish()
    }
}
